phpBB/includes/acp/acp_database.php

```diff
@@ -90,36 +90,11 @@ function main($id, $mode)
 						$time = time();
 
 						$filename = 'backup_' . $time . '_' . unique_id();
-						switch ($db->get_sql_layer())
-						{
-							case 'mysqli':
-							case 'mysql4':
-							case 'mysql':
-								$extractor = new mysql_extractor($format, $filename, $time, $download, $store);
-							break;
-
-							case 'sqlite':
-								$extractor = new sqlite_extractor($format, $filename, $time, $download, $store);
-							break;
-
-							case 'sqlite3':
-								$extractor = new sqlite3_extractor($format, $filename, $time, $download, $store);
-							break;
-
-							case 'postgres':
-								$extractor = new postgres_extractor($format, $filename, $time, $download, $store);
-							break;
 
-							case 'oracle':
-								$extractor = new oracle_extractor($format, $filename, $time, $download, $store);
-							break;
-
-							case 'mssql':
-							case 'mssql_odbc':
-							case 'mssqlnative':
-								$extractor = new mssql_extractor($format, $filename, $time, $download, $store);
-							break;
-						}
+						$extractor_factory = new \phpbb\db\extractor\factory();
```

Comment:
use the container here: `$phpbb_container->get('dbal.extractor');`

to do that you will have to change a little the extractor: don't use the constructor to get the parameters `$format, $filename, $time, $download, $store` but add an init() function for that (or a similar way)